network/topology.go

```diff
@@ -0,0 +1,51 @@
+package network
+
+import (
+	"fmt"
+	"sort"
+	"sync"
+	"sync/atomic"
+	"time"
+)
+
+type Topology interface {
+	MakeConnections(agents map[int]agentContainer)
+}
+
+type LinearTopology struct {
+	// number of nodes
+	NumNodes int
+}
+
+func (t LinearTopology) MakeConnections(agents map[int]agentContainer) {
+	keys := make([]int, 0)
+	for k := range agents {
+		keys = append(keys, k)
+	}
+	sort.Ints(keys)
+
+	startTime := time.Now()
+	wg := sync.WaitGroup{}
+	wg.Add(t.NumNodes)
```

Comment:
This should be a number of agents in agentCointainer

Reply:
Agree, number of wait groups is now length of agents map